### ISSUE TITLE
Feature request: specify config dir and current user at command line

### ISSUE THREAD

**Issue by jeffqg** [SomeOddCodeGuy/WilmerAI#4]
This is two feature requests, but they're so related, I'm putting them together.

1. Allow specifying config directly on the cli. This way users can copy the Public/Configs directory out of the repo directory and git pull without merging local workflow/endpoints/etc changes.
2. Allow specifying current_user on the cli. This is to make it easier to run multiple workflows from one installation. This could be simulated with the above FR, at the expense of maintaining multiple config directories.

**Comment by SomeOddCodeGuy** [comment]
I'll work on this over the weekend! I've also been working on something to help with the race condition for the workflow where summaries are created after responses. You use two models right? One for memories/summary, one for responses? If so, this will help a lot with it and speed up the general workflow a lot.

**Comment by SomeOddCodeGuy** [comment]
@jeffqg 

Ok, so while I wasn't able to get to this feature, I was able to get the initial version of a new feature to handle race conditions on with the chat summary change. It's in a new branch called feature/Workflow_Locks if you'd like to try it.

**tl;dr of the update:** _if you use the two model setup and are streaming: you no longer have to wait for summaries _at all_. They will simply happen in the background and you can keep chatting without thinking about them anymore. If you use the single model endpoint, it will simply stop you from sending messages to that endpoint while the summary is writing, and then you can send messages again when the summary is done._

I haven't updated the documentation yet, but here's how it works: WorkflowLock is a new node that you can put into a workflow, and for the instance of an endpoint call to the Wilmer API it will create a lock at that specific part of the workflow to ensure that if you hit that same point in the workflow a second time, you will get rejected. Once it is done with its work (or in the event of some failure) it releases the lock. If you turn wilmer off and back on, it will clear all old locks not belonging to that instance on load.

Let me give an example: In the single model convo/roleplay, I added a new custom workflow called FullCustomWorkflow-ChatSummary-WorkflowLocked. I put the lock at the start of that workflow. When you send a message, the workflow lock will be acquired and it'll respond to you, and then go off to do the summary; from your perspective, nothing different so far from what you're currently doing. However- since the workflow lock node is the very first node in the workflow, then until that summary is finished being written, Wilmer will just refuse any other attempts to run that workflow; the console will tell you its because a lock is in place on that workflow. This means you can't send any more messages until that summary is done.

Now, that's not super fancy but it does resolve the race condition issue. However, the real magic is if you use multiple models. For the two model convo/roleplay, it's way more fun. 

On the two model convo/roleplay user, I updated all workflows to use the "worker" endpoint for everything EXCEPT responding to you directly. I then made sure that the lock is set for _after_ the LLM responds to you. 

So it goes: step 1 is get file -> step 2 is respond to you -> step 3 is workflow lock -> step 4 is chat summary. Now when you send a message, it pulls the summary file (if it exists), responds to you, then locks, then does the summary.

With this change, because the responder endpoint is only used to respond and the worker is responsible for everything else, you could go ahead and send another message while the chat summary is being written, as the model doing all the memory/chat summary work is not the model that you'll use to respond. Your new message to wilmer back into the workflow, step 1 grabs the summary file (if it exists), step 2 responds, and then in step 3 the API call stops with an error. 

None of this affects the summary which is still working away in the background. Even if you cause the error to come up 3-4 times, the summary from your first call to the wilmer endpoint is still writing away in the background.

Using this, I was able to send a total of 4 messages in the time my summary was being written, then after the summary was complete and the file was updated, it released the lock. After that, my 5th message kicked off a new summary which re-locked the workflow, after which I sent more messages while that was being written. 

This is still VERY fresh from the oven; I loosely tested it while I worked and after I finished and so far it looks like it works great, but I haven't tested it to the level I'd feel comfortable pushing it to master yet. There's cleanup to do code-wise (I started to get into a hurry towards the end), but otherwise unless I find bugs then it's fully functional and "done" for now. I'll use it all week and hopefully can merge it later in the week or this weekend. Feel free to try it out yourself!

**Comment by SomeOddCodeGuy** [comment]
I haven't had it happen yet, but on the offchance it does: If you run into an issue of a lock not releasing, stop and restart Wilmer and it'll release all locks

**Comment by jeffqg** [comment]
The workflow lock sounds great. I'll give a try!

>  You use two models right? One for memories/summary, one for responses?

I do, but my reason has changed. I got a 3090 last week, which has been a game changer. I use RP tuned models most of the time, but I find they tend to either continue the narrative or hallucinate/misattribute messages. I've been trying out different instruct fine-tunes, and this evening was trying to see if I could send the memory creation to multiple models to be selected from.

**Comment by SomeOddCodeGuy** [comment]
@jeffqg First stab at implementing this issue is in PR #8 : https://github.com/SomeOddCodeGuy/WilmerAI/pull/8/

Please let me know if you run into any issues with it. It's still fresh out of the oven so I haven't had a chance to deeply test it yet.

**Comment by jeffqg** [comment]
I took a brief look at the code. I should have time to  try it out tomorrow. Thanks for adding this!
